### PR TITLE
Feat(logging): add logging package

### PR DESCRIPTION
**Pull request by sdwvit**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

File: packages/logging/src/errorTransformer.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ * @param fallback - original TraceKit.report method
+ * @returns Report, wrapped TraceKit.report method which no longer rethrows an error
+**/
+function safeWrapReport() {
+	function report(ex) {
+		try {
+			TraceKit.report(ex);
+		} catch (e) {
+			// TraceKit throws error bringing down current application, but we don't want that
+			// currently there is no way to fix it with some TraceKit option, thus monkey-patching
```
Comment (talend-glorieux) [inline packages/logging/src/errorTransformer.js]: Could you at least handle the exception by adding a console.error log or something?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

File: packages/logging/src/errorTransformer.js (hunk 1 above)
Comment (sdwvit) [inline packages/logging/src/errorTransformer.js]: sure, I can also replace with try finally if it bothers you

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

File: package.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "devDependencies": {
+    "babel-cli": "^6.23.0",
```
Comment (jmfrancois) [inline package.json]: babel-cli has nothing to do here

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

File: package.json (hunk 2 above)
Comment (sdwvit) [inline package.json]: the build is failing without it though

removed

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://208.talend.surge.sh)

**Pull request merged**